Doc/library/annotationlib.rst

```diff
@@ -0,0 +1,178 @@
+
+   *owner* is the object that owns the annotation function, usually a function, class, or module.
+   If provided, it is used in the :attr:`~Format.FORWARDREF` format to produce :class:`ForwardRef`
+   object that carry more information.
+
+   .. versionadded:: 3.14
+
+.. class:: Format
+
+   A :class:`enum.IntEnum` describing the formats in which annotations can be returned.
+   Members of the enum, or their equivalent integer values, can be passed to
+   :func:`get_annotations` and other functions in this module, as well as to
+   :attr:`~object.__annotate__` functions.
+
+   .. attribute:: VALUE
+
+      Values are the result of evaluating the annotation expressions.
+
+      This enum member's value is equal to 1.
+
+   .. attribute:: FORWARDREF
+
+      Values are real annotation values (as per :attr:`Format.VALUE` format) for defined values,
+      and :class:`ForwardRef` proxies for undefined values. Real objects may be exposed to,
+      or contain references to, :class:`ForwardRef` proxy objects.
+
+      This enum member's value is equal to 2.
+
+   .. attribute:: SOURCE
+
+      Values are the text string of the annotation as it appears in the source code.
+      May only be approximate; whitespace may be normalized, and constant values
+      may be optimized. It is possible the exact values of these strings could
+      change in future version of Python.
+
+   .. versionadded:: 3.14
+
+.. class:: ForwardRef
+
+   A proxy object for forward references in annotations.
+
+   Instances of this class are returned when the :attr:`~Format.FORWARDREF` format is
+   used and annotations contain a name that cannot be resolved.  This can happen
+   when a forward reference is used in an annotation, such as when a class is
+   referenced before it is defined.
+
+   .. attribute:: __forward_arg__
+
+      A string containing the code that was evaluated to produce the :class:`~ForwardRef`.
+      The string may not be exactly equivalent to the original source.
+
+   .. method:: evaluate(*, globals=None, locals=None, type_params=None, owner=None)
+
+      Evaluate the forward reference, returning its value.
+
+      This may throw an exception such as :exc:`NameError` if the forward reference
+      refers to names that do not exist. The parameters to the function can be used to
+      provide bindings for names that would otherwise be undefined.
+
+      *globals* and *locals* are passed to :func:`eval()`, representing the global and
+      local namespaces in which the name is evaluated. *type_params*, if given, must be
+      a tuple of :ref:`type parameters <type-params>` that are in scope while the forward
+      reference is being evaluated. *owner* is the object that owns the annotation from
+      which the forward reference derives, usually a function, class, or module.
+      :class:`~ForwardRef` instances returned by :func:`get_annotations` retain
```

Comment:
So, the *owner* parameter could somewhat be deprecated? or are there cases when it's really needed (or when we can use it for faking things?)

Reply:
The explicit parameter is relevant when ForwardRef objects are created other than by get_annotations(). For example, the typing module creates ForwardRef objects directly.

Reply:
Ok. So maybe for the user, it would just be clearer if you don't say that you don't need to pass them explicitly I think (or maybe say something along the lines "most of the time, you don't need to").

Reply:
Pushing an expansion of this paragraph.

Reply:
Thanks. Do you want me to review what you've pushed now or do you want to first correct all the occurrences of `()` inside `:func:` and usages of <code>\`\`</code> instead of `*`? (just ping me when you want a review/re-request a review)

Reply:
Thanks, I'll fix the rest of those too, and add documentation for some things I added in PRs that were just merged.